File: practica 1/sesion 3/DS_S3_C++Filters/CadenaFiltros.cpp
#include <exception>
using namespace std;

#include "CadenaFiltros.h"
#include "GestorFiltros.h"
#include "Filtro.h"
#include "Trabajo.h"

void CadenaFiltros::filtrar(Trabajo * unTrabajo) {
	vector<Filtro *>::iterator iter ;
	for (iter = listaFiltros.begin() ; iter != listaFiltros.end() ; iter++)
	{
		Filtro* filtro = *iter;
		filtro->filtrar(unTrabajo);
	}

	// Llamar en la cadena al target
	unTrabajo->getTecnico()->cobrar(unTrabajo->getPresupuesto());
}

File: practica 3/implementacion dart p3/lib/ModeloSOSEMADO/CadenaFiltros.dart
import 'GestorFiltros.dart';
import 'Trabajo.dart';
import 'Filtro.dart';

class CadenaFiltros{
  List<Filtro> listaFiltros = [];

  void filtrar(unTrabajo){
    listaFiltros.forEach((element){
      element.filtrar(unTrabajo);
    });
  }
}